### DISCUSSION TITLE
BCs around subdomains and visualization of result

### DISCUSSION THREAD

**Discussion by salaudeen-ya** [idaholab/moose#22470] [Q&A General] [answered]
Hello,

I am trying out MultiApp to solve two domains. In my simple case, the domains are solving the same equation since I am just literally splitting one domain into two.
The question is: I want to transfer between the two subdomains and I am not sure of the right boundary conditions to use. At the interface, I want right BC of the first one and the left BCs of the second domain to be Robin BCs (or a better way to specifiy the interfaceBC). Is there a way I can do that? The whole domain has DirichletBC on the left and Neumann on the right.

In addition, how can I visualize the result from the subdomains? By now, I am using Paraview option and I can only see the results of the subdomains separately. Is it possible to have a single plots from the subdomains plotted together? I mean, say the whole domain is 10 unit long and the interface is at point 5, how to plot from 0 - 10?

```
--------Parent---
`[Mesh]
  type = GeneratedMesh
  dim = 1
[]

[Variables]
  [u]
    order = FIRST
    family = LAGRANGE
  []
[]

[AuxVariables]
  [flux_u]
    order = CONSTANT
    family = MONOMIAL
  []
[]

[Kernels]
  [diffusion]
    type = MatDiffusion
    variable = u
    diffusivity = 2
  []
[]

[AuxKernels]
  [fluxKernel]
    type = DiffusionFluxAux
    diffusivity = 2
    variable = flux_u
    diffusion_variable = u
    component = x
  []
[]

[BCs]
  [leftBC] 
    type = DirichletBC
    variable = u
    boundary = 'left'
    value = 0
  []

  [rightBC] 
    type = NeumannBC
    variable = u
    boundary = 'right' 
    value = 1
  []
[]

[Executioner]
  type = Steady
  solve_type = 'PJFNK'

  fixed_point_max_its = 20
  nl_abs_tol = 1e-10
  fixed_point_rel_tol = 1e-16
  fixed_point_abs_tol = 1e-10

  petsc_options_iname = '-pc_type -pc_hypre_type'
  petsc_options_value = 'hypre boomeramg'
[]

[Outputs]
  exodus = true
  #csv = true
  console = true
[]


[MultiApps]
  [sub]
    type = FullSolveMultiApp
    execute_on = timestep_end
    positions = '0.5 0 0'
    input_files = sub.i
  []
[]

[Transfers]
  [pull_u]
    type = MultiAppNearestNodeTransfer
    to_multi_app = sub
    source_variable = u
    source_boundary = right
    variable = u_transfer
  []
[]`
```

```
------Subdomain------
`[Mesh]
  type = GeneratedMesh
  dim = 1
[]

[Variables]
  [u]
    order = FIRST
    family = LAGRANGE
  []
[]

[AuxVariables]
  [u_transfer]
    order = CONSTANT
    family = MONOMIAL
  []
[]

[Kernels]
  [diffusion]
    type = MatDiffusion
    variable = u
    diffusivity = 2
  []
[]

[AuxKernels]
  [fluxKernel]
    type = DiffusionFluxAux
    diffusivity = 2
    variable = u_transfer
    diffusion_variable = u
    component = x
  []
[]

[BCs]
  [leftBC] 
    type = DirichletBC
    variable = u
    boundary = 'left'
    value = 0
  []

  [rightBC] 
    type = DirichletBC
    variable = u
    boundary = 'right' 
    value = 1
  []
[]

[Executioner]
  type = Steady
  solve_type = 'PJFNK'

  petsc_options_iname = '-pc_type -pc_hypre_type'
  petsc_options_value = 'hypre boomeramg'
[]

[Outputs]
  exodus = true
  #csv = true
  console = true
[]
```

Thank you!

**Comment by GiudGiud** [comment, marked as answer]
Hello

for the plotting question, the easiest is to load multiple output files and display them together in Paraview. 

Note that you may translate simulations in MOOSE using the multi app positions parameter and/or modifying the origin of the app’s mesh (using a TransformGenerator for example)

you can transfer to an auxiliary variable using a nearest node strategy if the two domains don’t overlap. Then this variable will only contain the boundary value from the other application, and may be specified in a coupled boundary condition 

Guillaume
